src/tests/system/tests/test_proxy.py

```diff
@@ -67,3 +67,185 @@ def test_proxy__lookup_user_using_pam_ldap_and_nslcd_with_proxy_fast_alias_enabl
+    assert client.auth.ssh.password(f"{user2.name}@domain2.example.com", password="Secret123"), "User2 login failed"
+
+    # Verify distinct users by checking each user individually
+    user1_entry = client.tools.getent.passwd(f"{user1.name}@domain1.example.com")
+    user2_entry = client.tools.getent.passwd(f"{user2.name}@domain2.example.com")
+
+    assert user1_entry is not None and user2_entry is not None, "Expected two distinct users"
+    assert user1_entry.uid != user2_entry.uid, "Users have duplicate UIDs"
+
+
+@pytest.mark.importance("low")
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_proxy__case_preserving_handling(client: Client, ldap: LDAP):
+    """
+    :title: Proxy provider preserves original username case with case_sensitive=Preserving
+    :setup:
+        1. Create LDAP user "TestUser" with UID 5003 in ou=users
+        2. Configure SSSD with:
+           - id_provider=proxy
+           - proxy_lib_name=ldap
+           - case_sensitive=Preserving
+        3. Configure nslcd with:
+           - ignorecase=yes (for case-insensitive matching)
+           - validnames regex (to handle special characters)
+        4. Restart SSSD and nslcd services
+    :steps:
+        1. Perform user lookups with different case variants:
+           - getent passwd testuser
+           - getent passwd TESTUSER
+           - getent passwd TestUser
+        2. Verify authentication with different case variants:
+           - Authenticate as testuser
+           - Authenticate as TestUser
+           - Authenticate as TESTUSER
+    :expectedresults:
+        1. All case variants (testuser, TESTUSER, TestUser) should:
+           - Successfully match the LDAP user "TestUser"
+           - Return the original case ("TestUser") in responses
+           - Return correct home directory (/home/TestUser)
+        2. Authentication should succeed for all case variants
+    :customerscenario: False
+    """
+    # Setup
+    ou_users = ldap.ou("users").add()
+    ldap.user("TestUser", basedn=ou_users).add(uid=5003, gid=5003, password="Secret123", home="/home/TestUser")
+
+    # Configure SSSD with proxy provider
+    client.sssd.common.proxy("ldap", ["id", "auth"], server_hostname=ldap.host.hostname)
+    client.sssd.domain["case_sensitive"] = "Preserving"
+    client.sssd.svc.restart("nslcd")
+    client.sssd.restart()
+
+    client.fs.append(
+        "/etc/nslcd.conf",
+        "base dc=ldap,dc=test\n"
+        "ignorecase yes\n"
+        "validnames /^[a-z0-9._@$()]([a-z0-9._@$() ~-]*[a-z:0-9._@$()~-])?$/i\n",
+        dedent=False,
+    )
+    client.sssd.svc.restart("nslcd")
+    client.sssd.restart()
+
+    # Step 2: Test case preserving lookups
+    # All variants should match but preserve original case in output
+    for username in ["testuser", "TESTUSER", "TestUser"]:
+        result = client.tools.getent.passwd(username)
```

Comment:
Hi,

you have to restart SSSD with an empty cache for each attempt because otherwise the second and third lookup will just come from the cache.

bye,
Sumit

Reply:
Added

Reply:
Thanks